File: baekjoon/P10996/Main.java
package baekjoon.P10996;

import java.util.Scanner;

public class Main {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        int n = sc.nextInt();
        System.out.println(solve(n));
    }

    private static String solve(int n) {
        StringBuilder sb = new StringBuilder();

        for (int i = 0; i < n; i++) {
            sb.append(pattern(n, "*", " "));
            sb.append("\n");
            sb.append(pattern(n, " ", "*"));
            sb.append("\n");
        }

        return sb.toString();
    }

    private static String pattern(int n, String a, String b) {
        StringBuilder sb = new StringBuilder();

        for (int i = 0; i < n; i++) {
            if (i % 2 == 0) {
                sb.append(a);
            } else {
                sb.append(b);
            }
        }

        return sb.toString();
    }
}

File: baekjoon/P10996/Main.kt
package baekjoon.P10996

fun main() {
    val n = readLine()!!.toInt()
    println(solve(n))
}

private fun solve(n: Int): String {
    return buildString {
        repeat(n) {
            append(pattern(n, "*", " "))
            appendln()
            append(pattern(n, " ", "*"))
            appendln()
        }
    }
}

private fun pattern(n: Int, a: String, b: String): String {
    return buildString {
        for (i in 0 until n) {
            if (i % 2 == 0) {
                append(a)
            } else {
                append(b)
            }
        }
    }
}